swift/ArrowFlight/Tests/ArrowFlightTests/FlightTest.swift

```diff
@@ -0,0 +1,291 @@
+        client = FlightClient(channel: channel)
+    }
+    
+    deinit {
+        try? group?.syncShutdownGracefully()
+        try? channel?.close().wait()
+    }
+    
+    func listActionTest() async throws {
+        var actionTypes = [FlightActionType]()
+        try await client?.listActions( { action in
+            actionTypes.append(action)
+        })
+        
+        XCTAssertEqual(actionTypes.count, 2)
+        XCTAssertEqual(actionTypes[0].type, "type1")
+        XCTAssertEqual(actionTypes[0].description, "desc1")
+        XCTAssertEqual(actionTypes[1].type, "type2")
+        XCTAssertEqual(actionTypes[1].description, "desc2")
+    }
+    
+    func listFlightsTest() async throws {
+        let flightCriteria = FlightCriteria("flight criteria expression".data(using: .utf8)!)
+        var num_calls = 0
+        try await client?.listFlights(flightCriteria, closure: { data in
+            num_calls += 1
+            let schema = try streamToArrowSchema(data.schema)
+            XCTAssertEqual(schema.fields.count, 3)
+        })
+        
+        XCTAssertEqual(num_calls, 1)
+    }
+    
+    func doActionTest() async throws {
+        let action = FlightAction("test_action", body: "test_action body".data(using: .utf8)!)
+        var actionResults = [FlightResult]()
+        try await client?.doAction(action, closure: { result in
+            actionResults.append(result)
+        })
+        
+        XCTAssertEqual(actionResults.count, 1)
+        XCTAssertEqual(String(bytes:actionResults[0].body, encoding: .utf8), "test_action result")
+    }
+    
+    func getSchemaTest() async throws {
+        let descriptor = FlightDescriptor(cmd: "schema info".data(using: .utf8)!)
+        let schemaResult = try await client?.getSchema(descriptor)
+        let schema = try streamToArrowSchema(schemaResult!.schema)
+        XCTAssertEqual(schema.fields.count, 3)
+    }
+    
+    func doGetTest() async throws {
+        let ticket = FlightTicket("flight_ticket test".data(using: .utf8)!)
+        var num_call = 0
+        try await client?.doGet(ticket, readerResultClosure: { rb in
+            num_call += 1
+            XCTAssertEqual(rb.schema!.fields.count, 3)
+            XCTAssertEqual(rb.batches[0].length, 4)
+        })
+        
+        XCTAssertEqual(num_call, 1)
+    }
+    
+    func doPutTest() async throws {
+        let rb = try makeRecordBatch()
+        var num_call = 0
+        try await client?.doPut([rb], closure: { result in
+            num_call += 1
+        })
+        
+        XCTAssertEqual(num_call, 1)
+    }
+
+    func doExchangeTest() async throws {
+        let rb = try makeRecordBatch()
+        var num_call = 0
+        try await client?.doExchange([rb], closure: { result in
+            num_call += 1
+            XCTAssertEqual(result.schema?.fields.count, 3)
+            XCTAssertEqual(result.batches[0].length, 4)
+        })
+        
+        XCTAssertEqual(num_call, 1)
+    }
+}
+
+var serverup = false
```

Comment:
Can we move this to `testFlightServer()`?

Reply:
Will move variable into a Actor.